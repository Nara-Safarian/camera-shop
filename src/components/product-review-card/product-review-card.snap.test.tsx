import { render } from '@testing-library/react';
import ProductReviewCard from './product-review-card';

const review = {
  id: '1',
  createAt: 'Sat, 08 Jul 2023 11:58:27 GMT',
  cameraId: 1,
  userName: 'Test',
  advantage: 'Super',
  disadvantage: 'Not super',
  review: 'Cool thing. I suggest',
  rating: 4
};

test('ProductReviewCard snapshot should be rendered correctly', () => {
  const {container} = render(
    <ProductReviewCard review={review} />
  );
  expect(container).toMatchSnapshot();
});
