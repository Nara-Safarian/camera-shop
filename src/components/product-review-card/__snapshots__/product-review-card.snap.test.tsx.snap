// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductReviewCard snapshot should be rendered correctly 1`] = `
<div>
  <li
    class="review-card"
  >
    <div
      class="review-card__head"
    >
      <p
        class="title title--h4"
      >
        Test
      </p>
      <time
        class="review-card__data"
        datetime="2022-04-13"
      >
        Sat, 08 Jul 2023 11:58:27 GMT
      </time>
    </div>
    <div
      class="rate review-card__rate"
    >
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <p
        class="visually-hidden"
      >
        Оценка: 
        4
      </p>
    </div>
    <ul
      class="review-card__list"
    >
      <li
        class="item-list"
      >
        <span
          class="item-list__title"
        >
          Достоинства:
        </span>
        <p
          class="item-list__text"
        >
          Super
        </p>
      </li>
      <li
        class="item-list"
      >
        <span
          class="item-list__title"
        >
          Недостатки:
        </span>
        <p
          class="item-list__text"
        >
          Not super
        </p>
      </li>
      <li
        class="item-list"
      >
        <span
          class="item-list__title"
        >
          Комментарий:
        </span>
        <p
          class="item-list__text"
        >
          Cool thing. I suggest
        </p>
      </li>
    </ul>
  </li>
</div>
`;
